### PR TITLE
Implement Comment Model, CommentController, CommentService, and CommentRepository

### PR DESCRIPTION
**Pull request by janko101**
### CommentRepository
- Add Repository annotation
### Comment Model
- Add Spring annotations Entity, ID, GeneratedValue
- Add fields id, body, post
- Add ManyToOne relationship with the Post
- Code constructor, getters, setters
### CommentController
- Post Request for creating a comment for a specific post
- Get Request for returning all comments for a specific post
- Put Request for updating a comment by ID
- Delete Request for deleting a comment by ID
### Comment Service
- Create findCommentById method

### CODE REVIEW

File: src/main/java/se/kth/sda/skeleton/comments/Comment.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    @Column(nullable = false)
+    @NotBlank
+    private String body;
+
+    @ManyToOne
+    @JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
+    @JsonIdentityReference(alwaysAsId = true)
+    @JoinColumn(nullable = false)
+    @NotNull
+    private Post post;
```
Comment (fnt90) [inline src/main/java/se/kth/sda/skeleton/comments/Comment.java:28]: So, we will need to add corresponding @OneToMany on Post class

File: src/main/java/se/kth/sda/skeleton/exception/ResourceNotFoundException.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package se.kth.sda.skeleton.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.NOT_FOUND)
+public class ResourceNotFoundException extends RuntimeException {
+}
```
Comment (fnt90) [inline src/main/java/se/kth/sda/skeleton/exception/ResourceNotFoundException.java]: I notice this is inside package called "exception" , I think when Giang & I were working, I noticed a package called "exeption" . Was this a replacement?

File: src/main/java/se/kth/sda/skeleton/comments/Comment.java (hunk 1 above)
Comment (janko101) [reply src/main/java/se/kth/sda/skeleton/comments/Comment.java:28]: We already did...so you don't do it as it's out of your scope, but probably we'll have some merge issues once we merge your pr...

File: src/main/java/se/kth/sda/skeleton/exception/ResourceNotFoundException.java (hunk 2 above)
Comment (janko101) [reply src/main/java/se/kth/sda/skeleton/exception/ResourceNotFoundException.java]: No, we created also one. Can you just move yours from the "posts" package?

Comment (janko101) [reply src/main/java/se/kth/sda/skeleton/exception/ResourceNotFoundException.java]: Ideally, we should created a small PR before, where we add this exception, PostRepository and annotations in the Post model and then split from there, but what to do. We'll swim out of it somehow...

**Review by fnt90 — APPROVED** [review]
Looks great.

**Review by fnt90 — APPROVED** [review]
Yes good job

File: src/main/java/se/kth/sda/skeleton/comments/CommentService.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import se.kth.sda.skeleton.exception.ResourceNotFoundException;
+
+
+@Service
+public class CommentService {
+
+    CommentRepository commentRepository;
```
Comment (deciocandido) [inline src/main/java/se/kth/sda/skeleton/comments/CommentService.java:12]: I think the class CommentRepositoty should imported

File: src/main/java/se/kth/sda/skeleton/comments/CommentController.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
+package se.kth.sda.skeleton.comments;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import se.kth.sda.skeleton.exception.ResourceNotFoundException;
+import se.kth.sda.skeleton.posts.Post;
+import se.kth.sda.skeleton.posts.PostRepository;
+
```
Comment (deciocandido) [inline src/main/java/se/kth/sda/skeleton/comments/CommentController.java:10]: Import CommentRepository class and CommentService class

**Review by deciocandido — CHANGES_REQUESTED** [review]
There 2 suggestions to discuss, one on CommentService.java and the other on CommentController.java

**Pull request merged**